src/themes/default/components/core/AddToCart.vue

```diff
@@ -18,8 +18,8 @@ export default {
     onAfterRemovedVariant () {
       this.$forceUpdate()
     },
-    canBeAdded (product) {
-      return formatProductMessages(product.errors) !== ''
+    cantBeAdded (product) {
```

Comment:
this shouldn't be a method, but computed property `isDisabled` which returns Boolean value.
Also i think this.disabled should be first to check, then we don't have to invoke `formatProductMessages`